Exiled.API/Features/DamageHandlers/GenericDamageHandler.cs

```diff
@@ -0,0 +1,219 @@
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunAK);
+                    break;
+                case DamageType.MicroHid:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.MicroHID);
+                    break;
+                case DamageType.Explosion:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GrenadeHE);
+                    break;
+                case DamageType.Crossvec:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunCrossvec);
+                    break;
+                case DamageType.Logicer:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunLogicer);
+                    break;
+                case DamageType.Revolver:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunRevolver);
+                    break;
+                case DamageType.Shotgun:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunShotgun);
+                    break;
+                case DamageType.AK:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunAK);
+                    break;
+                case DamageType.Com15:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunCOM15);
+                    break;
+                case DamageType.Com18:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunCOM18);
+                    break;
+                case DamageType.Fsp9:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunFSP9);
+                    break;
+                case DamageType.E11Sr:
+                    GenericFirearm(player, attacker, damage, damageType, ItemType.GunE11SR);
+                    break;
+                case DamageType.ParticleDisruptor:
+                    Base = new DisruptorDamageHandler(Attacker, damage);
+                    break;
+
+                case DamageType.Scp096:
+                    PlayableScps.Scp096 curr096 = attacker.CurrentScp as PlayableScps.Scp096 ?? new PlayableScps.Scp096();
+                    if (curr096 != null)
+                    {
+                        curr096.Hub = attacker.ReferenceHub;
+                    }
+
+                    Base = new Scp096DamageHandler(curr096, damage, Scp096DamageHandler.AttackType.Slap);
+                    break;
+                case DamageType.Scp:
+                case DamageType.Scp018:
+                case DamageType.Scp207:
+                case DamageType.Scp049:
+                case DamageType.Scp173:
+                case DamageType.Scp939:
+                case DamageType.Scp0492:
+                case DamageType.Scp106:
+                    Base = new ScpDamageHandler(attacker.ReferenceHub, damage, DeathTranslations.Unknown);
```

Comment:
Why DeathTranslation Unknown?

Reply:
Fear/Laziness - I didn't want to specify for each type, and I didn't know whether we were going to keep this. I will add each individual type if you want